packages/components/src/Components/ConditionalFilter/Group.js

```diff
@@ -67,6 +66,8 @@ class Group extends Component {
                         e.stopPropagation();
                     }
 
+                    if (e.target.type === 'button' && onClick) { return onClick(); }
```

Comment:
I guess that the problem is with buttons and calling `onSelect` etc, how about we move this a bit down and check if target matches https://developer.mozilla.org/en-US/docs/Web/API/Element/matches instead of calling `type`?